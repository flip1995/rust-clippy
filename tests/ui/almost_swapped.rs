#![allow(clippy::needless_late_init, clippy::manual_swap)]
#![allow(unused_variables, unused_assignments)]
#![warn(clippy::almost_swapped)]

fn main() {
    let b = 1;
    let a = b;
    let b = a;

    let mut c = 1;
    let mut d = 2;
    d = c;
    c = d;

    let mut b = 1;
    let a = b;
    b = a;

    let b = 1;
    let a = 2;

    let t = b;
    let b = a;
    let a = t;

    let mut b = 1;
    let mut a = 2;

    let t = b;
    b = a;
    a = t;
}
