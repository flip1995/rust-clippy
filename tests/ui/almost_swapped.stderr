error: this looks like you are trying to swap `a` and `b`
  --> $DIR/almost_swapped.rs:7:5
   |
LL | /     let a = b;
LL | |     let b = a;
   | |______________^
   |
   = note: `-D clippy::almost-swapped` implied by `-D warnings`
   = note: maybe you could use `std::mem::swap(&mut a, &mut b)` or `std::mem::replace`?

error: this looks like you are trying to swap `d` and `c`
  --> $DIR/almost_swapped.rs:12:5
   |
LL | /     d = c;
LL | |     c = d;
   | |_________^ help: try: `std::mem::swap(&mut d, &mut c)`
   |
   = note: or maybe you should use `std::mem::replace`?

error: this looks like you are trying to swap `b` and `a`
  --> $DIR/almost_swapped.rs:16:5
   |
LL | /     let a = b;
LL | |     b = a;
   | |_________^
   |
   = note: maybe you could use `std::mem::swap(&mut b, &mut a)` or `std::mem::replace`?

error: aborting due to 3 previous errors

